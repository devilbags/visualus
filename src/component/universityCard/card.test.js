import React from 'react';
import { shallow } from 'enzyme';
import Card from './card';

describe('UniversityCard component', () => {
  const university = {
    id: 1,
    name: 'Boston University',
    alias: 'BU'
  };
  it('should render without crashing', () => {
    shallow(<Card university={university} />);
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<Card university={university} />);
    expect(wrapper).toMatchSnapshot();
  });
});
