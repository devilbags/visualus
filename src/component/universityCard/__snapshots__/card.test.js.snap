// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversityCard component with single data should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    university={
        Object {
            "admissionRate": 0.35,
            "alias": "BU",
            "annualCost": 96000,
            "city": "Boston",
            "id": 1,
            "name": "Boston University",
            "state": "MA",
            "studentSize": 1000,
            "website": "www.bu.edu",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <IconText
          text="Boston, MA"
          type="home"
/>,
        <IconText
          text="35%"
          type="usergroup-add"
/>,
        <IconText
          text="$96,000"
          type="shopping-cart"
/>,
        <IconText
          text="1,000"
          type="user"
/>,
        <IconText
          link="www.bu.edu"
          text="www.bu.edu"
          type="global"
/>,
      ],
      "children": <Meta
        avatar={
                <Avatar
                        prefixCls="ant-avatar"
                        shape="circle"
                        size="default"
                >
                        BU
                </Avatar>
        }
        title="Boston University"
/>,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "avatar": <Avatar
          prefixCls="ant-avatar"
          shape="circle"
          size="default"
>
          BU
</Avatar>,
        "title": "Boston University",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <IconText
            text="Boston, MA"
            type="home"
/>,
          <IconText
            text="35%"
            type="usergroup-add"
/>,
          <IconText
            text="$96,000"
            type="shopping-cart"
/>,
          <IconText
            text="1,000"
            type="user"
/>,
          <IconText
            link="www.bu.edu"
            text="www.bu.edu"
            type="global"
/>,
        ],
        "children": <Meta
          avatar={
                    <Avatar
                              prefixCls="ant-avatar"
                              shape="circle"
                              size="default"
                    >
                              BU
                    </Avatar>
          }
          title="Boston University"
/>,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "avatar": <Avatar
            prefixCls="ant-avatar"
            shape="circle"
            size="default"
>
            BU
</Avatar>,
          "title": "Boston University",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
